File: DSA_Level-2/Bit Manipulation/Copy Set Bits In A Range/CopySetBitsInARange_Solution.java
import java.io.*;
import java.util.*;

class Main {

  public static void main(String[] args){
    Scanner scn = new Scanner(System.in);
    int a = scn.nextInt();
    int b = scn.nextInt();
    int left = scn.nextInt();
    int right = scn.nextInt();

    int mask=(1<<(right-left+1))-1;
    mask=mask<<(left-1);
    
    mask=a&mask;
    mask=b|mask;
    
    System.out.println(mask);
    
  }

}

File: DSA_Level-2/Bit Manipulation/Copy Set Bits In A Range/CopySetBitsInARange_Solution.js

function copySetBits(a,b,n,m){
    
    let mask=1<<((m-n)-1);
    
    mask=mask<<(n-1);
    mask=mask&a;
    let res=b|mask;

    console.log(res);

}

copySetBits(10,13,2,3);